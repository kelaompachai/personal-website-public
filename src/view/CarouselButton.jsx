import React from 'react';
import Chevron from './Chevron';

function CarouselButton(props) {
  const { direction, carousel } = props;
  const fill = direction === 'left' ? <Chevron direction="left" /> : <Chevron direction="right" />;

  return (
    <button type="button" className="flex justify-center items-center size-24 border-2 border-well-read-500 rounded-3xl bg-well-read-200 hover:bg-well-read-300 active:transform active:translate-y-0.5" onClick={carousel}>{fill}</button>
  );
}

export default CarouselButton;
