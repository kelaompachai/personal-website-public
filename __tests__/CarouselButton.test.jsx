import React from 'react';
import { render, screen } from '@testing-library/react';
import CarouselButton from '../src/view/CarouselButton';


it('CarouselButton', () => {
  expect(render(<CarouselButton />).baseElement).toMatchSnapshot();

  // make sure there is a button
  const btn = screen.queryByRole('button');
  expect(btn).toBeInTheDocument();

  // make sure button is not disabled
  expect(btn).not.toBeDisabled();


  // check whether it goes down when pressed?
  // I think this is more of an end-to-end test--I don't know how to do it with RTL
});
