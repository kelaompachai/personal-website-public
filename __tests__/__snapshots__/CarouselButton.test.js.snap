// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselButton 1`] = `
<body>
  <div>
    <button
      class="flex justify-center items-center size-24 border-2 border-well-read-500 rounded-3xl bg-well-read-200 hover:bg-well-read-300 active:transform active:translate-y-0.5 cursor-pointer"
      type="button"
    >
      <svg
        class="size-1/2 fill-well-read-500"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</body>
`;
